### PR TITLE
Fix security context implementation

### PR DESCRIPTION
**Pull request by gigo1980**
**Issue**

https://github.com/gravitee-io/issues/issues/8314

**Description**

Fix the newline issue, that we have a valid kubernetes resource

### CODE REVIEW

**Comment by phiz71** [comment]
Hello @gigo1980 and thank you for your contribution ! May I ask you to add a unit-test ?

You can follow instructions in apim test folder: `apim/3.x/TESTS.md`

**Comment by remisultan** [comment]
👋 @gigo1980 and thanks for that first contribution

Would you be kind enough to test the behaviour in `ae/tests/deployment_test.yaml` ?

Here's what you could do.

```yaml
  - it: Deploy with custom security context
    template: engine-deployment.yaml
    set:
      engine:
        runAsNonRoot: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: "Always"
      - equal:
          path: spec.template.spec.containers[0].securityContext.runAsNonRoot
          value: true
      - equal:
          path: spec.template.spec.containers[0].securityContext.runAsUser
          value: 1001

  - it: Deploy without custom security context
    template: engine-deployment.yaml
    set:
      engine:
        runAsNonRoot: false
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: "Always"
      - isNull:
          path: spec.template.spec.containers[0].securityContext
```

Thank you !

**Comment by gigo1980** [comment]
@remisultan @phiz71:
Unittest that checks the right format for the security context has been added

File: ae/tests/deployment_test.yaml [hunk 1]
```diff
@@ -251,3 +251,28 @@ tests:
...
+      engine:
+        runAsNonRoot: true
+    asserts:
+      - hasDocuments:
+          count: 1
+      - isKind:
+          of: Deployment
+      - equal:
+          path: spec.template.spec.containers[0].securityContext.runAsNonRoot
+          value: true
```
Comment (remisultan) [inline ae/tests/deployment_test.yaml:278]: @gigo1980 could you also check the default `spec.template.spec.containers[0].securityContext.runAsUser` 

Thank you !

Comment (gigo1980) [reply ae/tests/deployment_test.yaml]: done

**Review by remisultan — APPROVED** [review]
LGTM

**Pull request merged**